core/src/ru/spbau/blackout/entities/DynamicObject.java

```diff
@@ -80,15 +86,34 @@ public void updateGraphics(float delta) {
     }
 
     @Override
-    public void updateForFirstStep() {
+    public void updateBeforeFirstStep() {
+        super.updateBeforeFirstStep();
+        // apply friction
+        if (!Utils.isZeroVec(velocity)){
```

Comment:
(optional) Use a http://wiki.c2.com/?GuardClause whenever the condition body is at least a few lines long. This case is borderline.